board/tests.py

```diff
@@ -10,16 +12,60 @@ class BoardModelTests(TestCase):
   A class with tests for testing relationships with the `Board` database model.
   """
 
+  def test_identical_photos(self):
+    """Make sure that photos with identical names and BLOBs can still be added to the image table."""
+    i1 = Image(name="shari", photo=bytearray("joseph", "utf-8"))
+    i2 = Image(name="shari", photo=bytearray("joseph", "utf-8"))
+    i1.save()
+    i2.save()
+
+    self.assertIsNot(i1, i2)
+
+  @tag('core')
+  def test_board_post_image_connections(self):
+    """Make sure that boards and posts and their connections to an image does not throw errors."""
+    b1 = Board(title="board1", description="bar")
+    b1.save()
+
+    i2 = Image(name="foo", photo=bytearray("bar", "utf-8"))
+    i3 = Image(name="ham", photo=bytearray("sam", "utf-8"))
+    i2.save()
+    i3.save()
+
+    b2 = Board(title="board2", description="foo", bg=i2)
+    b2.save()
+
+    self.assertIs(b2.bg, i2)
+    self.assertIs(i2.board, b2)
+
+    p1 = Post(associated_board=b2, name="p1", message="m1")
+    p2 = Post(associated_board=b2, name="p1", message="m1", photo=Image(name="spam", photo=bytearray("eggs", "utf-8")).save())
+    p3 = Post(associated_board=b2, name="p1", message="m1", photo=i3)
+    p1.save()
+    p2.save()
+    p3.save()
+
+    self.assertIs(p3.photo, i3)
+    self.assertIs(i3.post, p3)
+    self.assertIs(p2.photo.name, "spam")
```

Comment:
Can you write tests that throws exceptions and test those? Maybe I'll make a new issue for that.

Reply:
👌